### PR TITLE
[RFR] Place a marble on the board

### PR DESCRIPTION
**Pull request by xavierhans**
Create an empty board, display the board, ask for position and refresh the board game.

### CODE REVIEW

File: src/user.py [hunk 1]
```diff
@@ -14,7 +14,10 @@ def ask_line():
     return line
 
 
-def ask_position():
-    column = ask_column()
+def ask_position(board):
     line = ask_line()
-    print('{} {}'.format(column, line))
+    column = ask_column()
+    if board[line][column] != 'o':
```
Comment (fzaninotto) [inline src/user.py]: the value 'o' seems to mean EMPTY, so it's better to use a constant for that.

File: src/game.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+from user import ask_position
 
 
 def start_game():
-	board = create_empty_board()
-	display_board(board)
+    board = create_empty_board()
+    display_board(board)
+    new_position = ask_position(board)
+    board[new_position[0]][new_position[1]] = 'x'
```
Comment (fzaninotto) [inline src/game.py]: use a constant instead of a string

File: src/tests/test_board.py [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
-            [ 'o', 'o', 'o', 'o', 'o', 'o']])
+        self.assertEqual(create_empty_board(), [['o', 'o', 'o', 'o', 'o', 'o'],
+                                                ['o', 'o', 'o', 'o', 'o', 'o'],
+                                                ['o', 'o', 'o', 'o', 'o', 'o'],
+                                                ['o', 'o', 'o', 'o', 'o', 'o'],
+                                                ['o', 'o', 'o', 'o', 'o', 'o'],
+                                                ['o', 'o', 'o', 'o', 'o', 'o']])
 
 
 if __name__ == '__main__':
```
Comment (fzaninotto) [inline src/tests/test_board.py]: missing unit test for the feature you just added

File: Makefile [hunk 4]
```diff
@@ -7,10 +7,10 @@ install: ## Build the docker container
 	docker build -t pentago-python .
 
 run: ## Run the game
-	$(DOCKER) python3 main.py
+	$(DOCKER) python3 ./main.py
 
 test: ## Run the tests
-	$(DOCKER) python3 -m unittest discover -v
```
Comment (zyhou) [inline Makefile]: Add -v  `verbose` args

File: src/board_test.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import unittest
+from board import create_empty_board, NO_MARBLE
```
Comment (zyhou) [inline src/board_test.py]: `NO_MARBLE` from board ? I think constants

File: src/user.py [hunk 6]
```diff
@@ -1,20 +1,22 @@
...
-        print("The column number has to be a number between 0 and 5 included.")
-        return ask_column()
-    return column
+from constants import NO_MARBLE
 
 
-def ask_line():
-    line = int(input("Choose the line number of your next marble : "))
+def verify_position(board, line, column):
     if line < 0 or line > 5:
```
Comment (zyhou) [inline src/user.py]: Add 0 and 5 in constant

File: src/user.py [hunk 7]
```diff
@@ -1,20 +1,22 @@
...
-    line = int(input("Choose the line number of your next marble : "))
-    if line < 0 or line > 5:
+def verify_position(board, line, column):
+    if line < BOARD_START or line > BOARD_END:
         print("The line number has to be a number between 0 and 5 included.")
-        return ask_line()
-    return line
+        return False
+    if column < BOARD_START or column > BOARD_END:
+        print("The column number has to be a number between 0 and 5 included.")
```
Comment (fzaninotto) [inline src/user.py]: The idiomatic way to return several false status in Python is to create various Exception classes. Use try/catch on the caller side

File: src/user.py [hunk 8]
```diff
@@ -1,20 +1,22 @@
...
-    return column
+from constants import NO_MARBLE, BOARD_START, BOARD_END
 
 
-def ask_line():
-    line = int(input("Choose the line number of your next marble : "))
-    if line < 0 or line > 5:
+def verify_position(board, line, column):
+    if line < BOARD_START or line > BOARD_END:
         print("The line number has to be a number between 0 and 5 included.")
```
Comment (Kmaschta) [inline src/user.py]: You are mixing business logic and UI rendering.
Business logic is: a marble shouldn't be outside of the board.
The UI issue is: if there is a marble outside of board error, the console should display `The line number has to be a number between 0 and 5 included.`.

IMHO, it's better to isolate both in order to test them both.
So the best way to do this is to have dedicated exceptions (like `OutOfBoardMarbleException` or so) and the UI should handle these exceptions to display messages.

**Comment by xavierhans** [comment]
Ready for review

**Pull request merged**